Bug 1837998 - File regression bugs using the API

junngo opened the pull request:
Hi there.
I delete the `cf_has_regression_range` field. I guess, This field was deprecated.
Before We filed the bug without the log-in, but  We should log in to call the post request and to file the bug at the moment.
Feel free to tell me If there is something to be modified :)

bug link: https://bugzilla.mozilla.org/show_bug.cgi?id=1837998

gmierz commented:
@junngo and I discussed this patch yesterday, and one thing to change will be to output a failure message if the bug creation doesn't work.

beatrice-acasandrei commented:
I think it's looking great, I wasn't able to test the code locally yet, I have some issues downloading data in order to do so.  Perhaps we can do a deploy to staging in order for all the sheriffs to test this. What do you think @gmierz @junngo?

junngo commented on treeherder/webapp/api/bugzilla.py:
```diff
@@ -34,7 +34,10 @@ def create_bug(self, request):
         ).encode("utf-8")
         summary = params.get("summary").encode("utf-8").strip()
         url = settings.BUGFILER_API_URL + "/rest/bug"
-        headers = {"x-bugzilla-api-key": settings.BUGFILER_API_KEY, "Accept": "application/json"}
+        headers = {
+            "x-bugzilla-api-key": settings.BUGFILER_API_KEY,
```
FYI:
I set the buzilla api url[0] and api key[1] for the test through hardcoding on my local.

[0] BUGFILER_API_URL: "https://bugzilla.mozilla.org"
[1] BUGFILER_API_KEY: "My bugzilla api key"

junngo commented:
I added the feature. If there is an error from the server, we can see the error message in the modal.
Let me know If there is anything to change :)

@beatrice-acasandrei @gmierz

beatrice-acasandrei commented:
@junngo looks good! If you need help with deploying to staging feel free to ping me on element!

junngo commented on ui/perfherder/alerts/StatusDropdown.jsx:
```diff
@@ -200,6 +206,12 @@ export default class StatusDropdown extends React.Component {
+      this.setState({
+        fileBugErrorMessage: null,
+      });
+    }
```
I added a code to clear the error message when closing the file bug modal.

beatrice-acasandrei commented on ui/perfherder/alerts/StatusDropdown.jsx:
```diff
@@ -61,7 +57,11 @@ export default class StatusDropdown extends React.Component {
+    let needinfoFrom = '';
+    if (bugData.assigned_to !== 'nobody@mozilla.org') {
+      needinfoFrom = bugData.assigned_to;
+    }
```
If there's no asignee on the ticket then we should set a needinfo for the triage owner.

beatrice-acasandrei replied:
There's also another edge case when the asignee is disabled (here's an [example](https://bugzilla.mozilla.org/show_bug.cgi?id=907925) of an old bug). Just to be safe I think it's best to also check if the user is disabled - if so, then the needinfo should be set to the triage owner.

beatrice-acasandrei replied:
Hi @junngo ! I'll try to clarify your question if I understand correctly what you are asking. The creator(reporter) is a sheriff (including me and other team members), which is different from the person that worked on the patch. By setting a needinfo we are requesting to the person that caused a certain regression to look into it and potentially backout their changes. For more context adding a person to the needinfo field is an esential step in the sheriffing process. If we don't set a needinfo automatically then the sheriff needs to go over the created bug and add it manually which is the oposite of what we want to achieve here.

beatrice-acasandrei replied:
@junngo could you investigate if there's a way to get the details of a certain component in bugzilla? every bug has a component. The component details should contain the triage owner's information (id or something similar).
I've seen that they used this endpoint in the UI https://bugzilla.mozilla.org/rest/component-watching?Bugzilla_api_token=jjt9LqN7XjuFLJUftBra9x so that's were I got the idea.

junngo replied:
Thanks for reviewing :)
I add the triage owner If there is no assignee.
@beatrice-acasandrei @gmierz

junngo commented:
I had to set and check the api key and URL for the test in my local environment.

- Add the BUGFILER_API_KEY
   - Move the Bugzilla to generate the api key (https://bugzilla.mozilla.org/userprefs.cgi?tab=apikey)
   - Add the api key into the production `env` file[0] to call the api[1]. In my case I added the api key at the source code[1] in person instead of using the env file.
- Check the BUGFILER_API_URL
   - We need the api URL[2] to call the api of buazilla. That is the `https://bugzilla.mozilla.org`. We should check If the URL[3] is set in the `env` file. I added the api URL in person at the source code[2] instead of using env file.
- Ready for the testing. Move the perfherder and Click the `File Bug` to test it.

Let me know If you have a question :)

[0]
https://github.com/mozilla/treeherder/blob/6c6407b2c70d117493319895faf2138ed620d1c1/treeherder/config/settings.py#L436
[1]
https://github.com/mozilla/treeherder/blob/6c6407b2c70d117493319895faf2138ed620d1c1/treeherder/webapp/api/bugzilla.py#L37
[2]
https://github.com/mozilla/treeherder/blob/6c6407b2c70d117493319895faf2138ed620d1c1/treeherder/webapp/api/bugzilla.py#L36
[3]
https://github.com/mozilla/treeherder/blob/6c6407b2c70d117493319895faf2138ed620d1c1/treeherder/config/settings.py#L435

beatrice-acasandrei reviewed (approved):
@gmierz  @junngo I've tested it locally and I had no issues, thanks. I'll also show the functionality to @esanuandra @florinbilt and Alex Finder tomorrow to make sure there's nothing I'm missing and we can merge this.

beatrice-acasandrei commented:
I did some more testing based on the discussion with the Softvision sheriffing team and what I think the following would need to be updated:

- [ ] When we don't introduce any bug number there should still be a needinfo to the triage owner so that we're always aware of the bugs being created, even by mistake.
- [ ] For devtools I couldn't file a bug, the keyword _devtools-perf-regression_ seems to be wrong 
- [ ] For mozperftest I get _Typeerror: result is undefined_

Let me know what you think.

gmierz commented:
@beatrice-acasandrei the keyword has been added: https://bugzilla.mozilla.org/show_bug.cgi?id=1729635#c7

junngo commented:
Hi @beatrice-acasandrei, @gmierz  :)
I'm sorry for not getting back to you sooner.
I changed the code to add the needinfo to the default triage owner If there is no entered bug.

Who is the needinfo?
| Case                                                      | who                                        |
|-----------------------------------------------------|----------------------------------------|
| No Entered Bug                                     | default triage owner               |
| If there is no assignee on entered bug  | triage owner on entered bug  |
| If there is assignee on the entered bug | assignee on entered bug       | 

Let me know if anything needs to be changed.
